openfisca_aotearoa/variables/acts/social_security/sole_parent_support/sole_parent_support.py

```diff
@@ -105,33 +118,207 @@ class sole_parent_support__age_threshold(variables.Variable):
     entity = entities.Person
-    label = "Has lived continuously in New Zealand for 2 years or more at any one time since becoming a New Zealand citizen or permanent resident?"
-    definition_period = periods.MONTH
-    reference = "TODO"
+    label = "Income is below Sole Parent Support threshold?"
+    definition_period = periods.WEEK
+    set_input = holders.set_input_divide_by_period
 
-    def formula(persons, period, parameters):
-        # been in NZ log enough?
-        min_years = parameters(period).entitlements.social_security.sole_parent_support.minumum_continuous_time_in_nz
-        years_in_nz = persons("number_of_years_lived_in_nz", period)
-        return years_in_nz >= min_years
 
+class sole_parent_support__requirement(variables.Variable):
+    value_type = bool
+    default_value = False
+    entity = entities.Person
+    label = "Meets the sole parent requirement?"
+    definition_period = periods.WEEK
+    reference = "https://www.legislation.govt.nz/act/public/2018/0032/latest/whole.html#DLM6783167"
+    set_input = holders.set_input_dispatch_by_period
 
-class sole_parent_support__below_income_threshold(variables.Variable):
+    def formula_2018_11_26(persons, period, parameters):
+
+        ssa30_dependent_child_under_14 = persons("sole_parent_support__dependent_child_requirement", period, "add")
+        # ssa30 = ssa30_person_is_parent * ssa30_person_has_dependent_child * ssa30_dependent_child_under_14
+
+        ssa30_a = logical_not(persons("social_security__in_a_relationship", period))
+
+        ssa30_b = persons("sole_parent_support__spouse_or_partner_died", period)
+
+        ssa30_c = persons("sole_parent_support__marriage_or_civil_union_dissolved", period)
+
+        ssa30_d = persons("sole_parent_support__living_apart_and_lost_support", period)
+
+        ssa30_e = persons("sole_parent_support__lost_regular_support", period)
+
+        return ssa30_dependent_child_under_14 * (ssa30_a + ssa30_b + ssa30_c + ssa30_d + ssa30_e)
+
+
+class sole_parent_support__dependent_child_requirement(variables.Variable):
     value_type = bool
-    default_value = True
+    default_value = False
     entity = entities.Person
-    label = "Income is below Sole Parent Support threshold?"
-    definition_period = periods.MONTH
+    label = "At least one dependent child meets age requirement"
+    definition_period = periods.WEEKDAY
+    reference = "https://www.legislation.govt.nz/act/public/2018/0032/latest/whole.html#DLM6783167"
+
+    def formula_2018_11_26(persons, period, parameters):
+        child_age_threshold = parameters(period).social_security.sole_parent_support.child_age_threshold
+
+        dc = persons.family.members("social_security__dependent_child", period.first_week) * \
+            persons.family.members("age", period.first_day) < child_age_threshold
+
+        return persons.family.any(dc, role=entities.Family.CHILD) * (persons.has_role(entities.Family.PRINCIPAL) + persons.has_role(entities.Family.PARENT))
```

Comment:
Interesting.

I would have thought you were going to use _social_security__principal_caregiver_ for this:

```suggestion
        principal = people("social_security__principal_caregiver", period)
        children = persons.family.any(dc, role=entities.Family.CHILD)
        parent = persons.has_role(entities.Family.PARENT)
        return principal * children * parent
```